`test_oversubscribing_leases` flaky

On macOS, 3.7: https://github.com/dask/distributed/runs/2947008026#step:10:2421

```python
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

c = <Client: No scheduler connected>
s = <Scheduler: "tcp://127.0.0.1:56382" workers: 0 cores: 0, tasks: 0>
a = <Worker: 'tcp://127.0.0.1:56383', 0, Status.closed, stored: 0, running: 0/1, ready: 0, comm: 0, waiting: 0>
b = <Worker: 'tcp://127.0.0.1:56387', 1, Status.closed, stored: 0, running: 0/2, ready: 0, comm: 0, waiting: 0>

    @pytest.mark.flaky(reruns=10, reruns_delay=5, condition=WINDOWS)
    @gen_cluster(
        client=True,
        config={
            "distributed.scheduler.locks.lease-timeout": "100ms",
            "distributed.scheduler.locks.lease-validation-interval": "100ms",
        },
    )
    async def test_oversubscribing_leases(c, s, a, b):
        """
        This test ensures that we detect oversubscription scenarios and will not
        accept new leases as long as the semaphore is oversubscribed.
    
        Oversubscription may occur if tasks hold the GIL for a longer time than the
        lease-timeout is configured causing the lease refresh to go stale and timeout.
    
        We cannot protect ourselves entirely from this but we can ensure that while
        a task with a timed out lease is still running, we block further
        acquisitions until we return to normal.
    
        An example would be a task which continuously locks the GIL for a longer
        time than the lease timeout but this continuous lock only makes up a
        fraction of the tasks runtime.
        """
        # GH3705
    
        from distributed.worker import Worker, get_client
    
        # Using the metadata as a crude "asyncio.Event" since the proper event
        # implementation cannot be serialized. For the purpose of this test a
        # metadata check with a sleep loop is not elegant but practical.
        await c.set_metadata("release", False)
        sem = await Semaphore()
        sem.refresh_callback.stop()
    
        def guaranteed_lease_timeout(x, sem):
            """
            This function simulates a payload computation with some GIL
            locking in the beginning.
    
            To simulate this we will manually disable the refresh callback, i.e.
            all leases will eventually timeout. The function will only
            release/return once the "Event" is set, i.e. our observer is done.
            """
            sem.refresh_leases = False
            client = get_client()
    
            with sem:
                # This simulates a task which holds the GIL for longer than the
                # lease-timeout. This is twice the lease timeout to ensurre that the
                # leases are actually timed out
                slowidentity(delay=0.2)
    
                assert sem._leases
                # Now the GIL is free again, i.e. we enable the callback again
                sem.refresh_leases = True
                sleep(0.1)
    
                # This is the poormans Event.wait()
                while client.get_metadata("release") is not True:
                    sleep(0.05)
    
                assert sem.get_value() >= 1
                return x
    
        def observe_state(sem):
            """
            This function is 100% artificial and acts as an observer to verify
            our assumptions. The function will wait until both payload tasks are
            executing, i.e. we're in an oversubscription scenario. It will then
            try to acquire and hopefully fail showing that the semaphore is
            protected if the oversubscription is recognized.
            """
            sem.refresh_callback.stop()
            # We wait until we're in an oversubscribed state, i.e. both tasks
            # are executed although there should only be one allowed
            while not sem.get_value() > 1:
                sleep(0.2)
    
            # Once we're in an oversubscribed state, we must not be able to
            # acquire a lease.
            assert not sem.acquire(timeout=0)
    
            client = get_client()
            client.set_metadata("release", True)
    
        observer = await Worker(s.address)
    
        futures = c.map(
            guaranteed_lease_timeout, range(2), sem=sem, workers=[a.address, b.address]
        )
        fut_observe = c.submit(observe_state, sem=sem, workers=[observer.address])
    
        with captured_logger("distributed.semaphore", level=logging.DEBUG) as caplog:
            payload, observer = await c.gather([futures, fut_observe])
    
        logs = caplog.getvalue().split("\n")
        timeouts = [log for log in logs if "timed out" in log]
        refresh_unknown = [log for log in logs if "Refreshing an unknown lease ID" in log]
>       assert len(timeouts) == 2
E       assert 4 == 2
E         +4
E         -2

distributed/tests/test_semaphore.py:440: AssertionError
```
This is already marked as flaky on windows, maybe it should be marked flaky everywhere?

--- a/distributed/tests/test_semaphore.py
+++ b/distributed/tests/test_semaphore.py
@@ -338,7 +338,6 @@ async def test_retry_acquire(c, s, a, b):
         assert result is False
 
 
-@pytest.mark.flaky(reruns=10, reruns_delay=5, condition=WINDOWS)
 @gen_cluster(
     client=True,
     config={
@@ -437,12 +436,13 @@ async def test_oversubscribing_leases(c, s, a, b):
     logs = caplog.getvalue().split("\n")
     timeouts = [log for log in logs if "timed out" in log]
     refresh_unknown = [log for log in logs if "Refreshing an unknown lease ID" in log]
-    assert len(timeouts) == 2
-    assert len(refresh_unknown) == 2
+    assert len(timeouts) >= 2
+    assert len(refresh_unknown) >= 2
 
     assert sorted(payload) == [0, 1]
     # Back to normal
-    assert await sem.get_value() == 0
+    while await sem.get_value():
+        await asyncio.sleep(0.01)
 
 
 @gen_cluster(client=True)

